ts/data.ts

```diff
@@ -1,8 +1,40 @@
-/* exported data */
-
-let data = {
-  view: 'entry-form',
-  entries: [],
-  editing: null,
-  nextEntryId: 1,
-};
+/* exported data, writeData */
+
+interface Entry {
+  title: string;
+  photoUrl: string;
+  notes: string;
+  entryId: number;
+}
+
+interface DataObject {
+  view: string;
+  entries: Entry[];
+  editing: null;
+  nextEntryId: number;
+}
+
+const data = readData();
+
+writeData();
+
+function writeData(): void {
+  const dataJson = JSON.stringify(data);
+  localStorage.setItem('entriesData', dataJson);
+}
+
+function readData(): DataObject {
+  if (!localStorage.getItem('entriesData')) {
+    const dataDefault: DataObject = {
+      view: 'entry-form',
+      entries: [] as Entry[],
+      editing: null,
+      nextEntryId: 1,
+    };
+    return dataDefault;
+  }
+
+  const dataJson = localStorage.getItem('entriesData') as string;
+  const dataObj = JSON.parse(dataJson);
+  return dataObj;
```

Comment:
I would place your line 37 above your `if` statement and remove the type assertion. This will fix the issue where you needed type assertion and keep you from retrieving the data twice. You will also need to change your condition of you `if` statement. (Your `if` statement on line 27 is running the same search as on line 37, we want to try and avoid repeating actions).